main/src/cgeo/geocaching/utils/formulas/Formula.java

```diff
@@ -408,7 +408,8 @@ protected void doCompile(final String rawExpression, final int startPos, final F
             throw new FormulaException(EMPTY_FORMULA);
         }
         final FormulaNode x = parseExpression();
-        if (!p.eof()) {
+        // line end or beginning of user comment
+        if (!p.eof() && !p.eat('#')) {
```

Comment:
I understand what you want to reach, but I believe if the code is applied like this then there might be some unwanted side effects in calculated coordinate parser (which uses concatennated Formula parsing instances) in case # is used there

E.g. something like this might actually get parsed: `N48 12#comment.345`.

But I haven't tried this out

Reply:
Formula class must set the Parser position to the next character AFTER the parsed formula and save in the expression the exact part of the string which was parsed from rawExpression (if stopChecker is used then the epression is not parsed until its end always, this is heavily used by DegreeFormula). 

I think your code will not put the parser position at the end of the parsed part of the rawExpression. It aborts when it finds #, leaving the parser position on the #. Parsing must continue until end of expression

Additionally I would allow also to close a comment again (e.g. by a second #) this would allow commenting also inside coordinate expressions without harming the parsing.

Finally, your solution only allows comments on the END of a parsed formula expression and not somewhere inbetween. For example `3.14 #this is pi# * R #this is the radius used # * 2` would not work